diff.py

```diff
@@ -475,10 +476,71 @@ def __init__(self, value: int, ansi_code: str):
 FormatFunction = Callable[[str], Format]
 
 
+@dataclass
+class Text:
+    segments: List[Tuple[str, Format]]
+
+    @staticmethod
+    def formatted(line: str, f: Format) -> "Text":
+        return Text([(line, f)])
+
+    @staticmethod
+    def raw(line: str) -> "Text":
+        return Text.formatted(line, Format.NONE)
+
+    @staticmethod
+    def empty() -> "Text":
+        return Text([])
```

Comment:
having a single constructor `def __init__(self, line: Optional[str] = None, f: Format = Format.NONE)` might look nicer

Reply:
Yeah I think this makes it *a lot* nicer! Good call!